contracts/eosio.system/delegate_bandwidth.cpp

```diff
@@ -0,0 +1,261 @@
+/**
+ *  @file
+ *  @copyright defined in eos/LICENSE.txt
+ */
+#include "eosio.system.hpp"
+
+#include <eosiolib/eosio.hpp>
+#include <eosiolib/print.hpp>
+#include <eosiolib/datastream.hpp>
+#include <eosiolib/serialize.hpp>
+#include <eosiolib/multi_index.hpp>
+#include <eosiolib/privileged.h>
+#include <eosiolib/transaction.hpp>
+
+#include <eosio.token/eosio.token.hpp>
+
+#include <map>
+
+namespace eosiosystem {
+   using eosio::asset;
+   using eosio::indexed_by;
+   using eosio::const_mem_fun;
+   using eosio::bytes;
+   using eosio::print;
+   using eosio::permission_level;
+   using std::map;
+   using std::pair;
+
+   static constexpr time refund_delay = 3*24*3600;
+   static constexpr time refund_expiration_time = 3600;
+
+   struct total_resources {
+      account_name  owner;
+      asset         net_weight;
+      asset         cpu_weight;
+      asset         storage_stake;   
+      uint64_t      storage_bytes = 0;
+
+      uint64_t primary_key()const { return owner; }
+
+      EOSLIB_SERIALIZE( total_resources, (owner)(net_weight)(cpu_weight)(storage_stake)(storage_bytes) )
```

Comment:
Is this required or does it work without?

Reply:
How would the multi_index tables below use if these were not defined?